ros2_foxglove_bridge/src/ros2_foxglove_bridge_node.cpp

```diff
@@ -3,27 +3,31 @@
 int main(int argc, char* argv[]) {
   rclcpp::init(argc, argv);
 
-  auto dummyNode = std::make_shared<rclcpp::Node>("dummy");
-  auto numThreadsDescription = rcl_interfaces::msg::ParameterDescriptor{};
-  numThreadsDescription.name = "num_threads";
-  numThreadsDescription.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
-  numThreadsDescription.description =
-    "The number of threads to use for the ROS node executor. 0 means one thread per CPU core.";
-  numThreadsDescription.read_only = true;
-  numThreadsDescription.additional_constraints = "Must be a non-negative integer";
-  numThreadsDescription.integer_range.resize(1);
-  numThreadsDescription.integer_range[0].from_value = 0;
-  numThreadsDescription.integer_range[0].to_value = INT32_MAX;
-  numThreadsDescription.integer_range[0].step = 1;
-  constexpr int DEFAULT_NUM_THREADS = 0;
-  dummyNode->declare_parameter("num_threads", DEFAULT_NUM_THREADS, numThreadsDescription);
-
-  const auto numThreads = dummyNode->get_parameter("num_threads").as_int();
+  size_t numThreads = 0;
+  {
+    // Temporary dummy node to get num_threads param.
+    auto dummyNode = std::make_shared<rclcpp::Node>("dummy");
+    auto numThreadsDescription = rcl_interfaces::msg::ParameterDescriptor{};
+    numThreadsDescription.name = "num_threads";
+    numThreadsDescription.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    numThreadsDescription.description =
+      "The number of threads to use for the ROS node executor. 0 means one thread per CPU core.";
+    numThreadsDescription.read_only = true;
+    numThreadsDescription.additional_constraints = "Must be a non-negative integer";
+    numThreadsDescription.integer_range.resize(1);
+    numThreadsDescription.integer_range[0].from_value = 0;
+    numThreadsDescription.integer_range[0].to_value = INT32_MAX;
+    numThreadsDescription.integer_range[0].step = 1;
+    constexpr int DEFAULT_NUM_THREADS = 0;
+    dummyNode->declare_parameter(numThreadsDescription.name, DEFAULT_NUM_THREADS,
+                                 numThreadsDescription);
+    numThreads = static_cast<size_t>(dummyNode->get_parameter(numThreadsDescription.name).as_int());
+  }
 
   auto executor =
     rclcpp::executors::MultiThreadedExecutor::make_shared(rclcpp::ExecutorOptions{}, numThreads);
 
-  rclcpp_components::ComponentManager componentManager(executor, "ComponentManager");
+  rclcpp_components::ComponentManager componentManager(executor, "foxglove_component_manager");
```

Comment:
This may need to be `foxglove_bridge_component_manager` if it's supposed to be globally unique in the rosgraph, since we may introduce more Foxglove ROS nodes over time.

Reply:
Changed during rebasing. Since #125 we could actually get rid of the executable, since one can load the compoment directly in a launch file: https://docs.ros.org/en/humble/How-To-Guides/Launching-composable-nodes.html